src/sinks/splunk_hec/common/service.rs

```diff
@@ -39,18 +40,22 @@ struct HecAckResponseBody {
     ack_id: Option<u64>,
 }
 
-impl HecService {
+impl<S> HecService<S>
+where
+    S: Service<HecRequest> + Send + 'static,
+    S::Future: Send + 'static,
+    S::Response: Response + ResponseExt + Send + 'static,
+    S::Error: fmt::Debug + Into<crate::Error> + Send,
+{
     pub fn new(
-        client: HttpClient,
-        http_request_builder: HttpRequestBuilder,
+        inner: S,
+        ack_client: Option<HttpClient>,
+        http_request_builder: Arc<HttpRequestBuilder>,
         indexer_acknowledgements: HecClientAcknowledgementsConfig,
     ) -> Self {
-        let event_client = client.clone();
-        let ack_client = client;
-        let http_request_builder = Arc::new(http_request_builder);
         let max_pending_acks = indexer_acknowledgements.max_pending_acks.get();
-        let tx = if indexer_acknowledgements.indexer_acknowledgements_enabled {
-            let (tx, rx) = mpsc::channel(indexer_acknowledgements.max_pending_acks.get() as usize);
+        let tx = if let Some(ack_client) = ack_client {
+            let (tx, rx) = mpsc::channel(max_pending_acks as usize);
```

Comment:
This seems way too big.

You only need enough space to allow a constant stream of pending ack oneshots to be sent to the acknowledgements task, but it doesn't need to actually hold the maximum number of pending acks at the very same time.

This should probably be more like 64-128?

Reply:
Gotcha, makes sense. Adjusted to `128`.